### PR TITLE
[@hapi/jwt] type definitions

### PR DESCRIPTION
**Pull request by ssanchezmarc**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by devinstewart — APPROVED** [review]
As a maintainer of the [@hapi/jwt](https://github.com/hapijs/jwt) package, this looks good.   I appreciate the work done by @ssanchezmarc as we are not able to create types directly in the module due to its tight integration with [@hapi/hapi](https://github.com/hapijs/hapi), which is also located here.

File: types/hapi__jwt/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,442 @@
+// Type definitions for @hapi/jwt 2.0
+// Project: https://github.com/hapijs/jwt
+// Definitions by: Sergio Sánchez <https://github.com/ssanchezmarc>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 4.2
```
Comment (amcasey) [inline types/hapi__jwt/index.d.ts:5]: Why was this required?

**Review by amcasey — APPROVED** [review]

File: types/hapi__jwt/index.d.ts (hunk 1 above)
Comment (ssanchezmarc) [reply types/hapi__jwt/index.d.ts:5]: I try to follow the same conventions I saw in other similar cases where we have defined types for hapi modules. No sure if it is mandatory or just optional :)

**Comment by ssanchezmarc** [comment]
Ready to merge
